data/helpers.d/php

```diff
@@ -65,3 +149,235 @@ ynh_remove_fpm_config () {
 	ynh_secure_remove --file="$fpm_config_dir/conf.d/20-$app.ini" 2>&1
 	ynh_systemd_action --service_name=$fpm_service --action=reload
 }
+
+# Install another version of php.
+#
+# usage: ynh_install_php --phpversion=phpversion [--package=packages]
+# | arg: -v, --phpversion - Version of php to install.
+# | arg: -p, --package - Additionnal php packages to install
+ynh_install_php () {
+	# Declare an array to define the options of this helper.
+	local legacy_args=vp
+	declare -Ar args_array=( [v]=phpversion= [p]=package= )
+	local phpversion
+	local package
+	# Manage arguments with getopts
+	ynh_handle_getopts_args "$@"
+	package=${package:-}
+
+	# Store phpversion into the config of this app
+	ynh_app_setting_set $app phpversion $phpversion
+
+	if [ "$phpversion" == "7.0" ]
+	then
+		ynh_die "Do not use ynh_install_php to install php7.0"
+	fi
+
+	# Store the ID of this app and the version of php requested for it
+	echo "$YNH_APP_INSTANCE_NAME:$phpversion" | tee --append "/etc/php/ynh_app_version"
+
+	# Add an extra repository for those packages
+	ynh_install_extra_repo --repo="https://packages.sury.org/php/ $(lsb_release -sc) main" --key="https://packages.sury.org/php/apt.gpg" --priority=995 --name=extra_php_version
+
+	# Install requested dependencies from this extra repository.
+	# Install php-fpm first, otherwise php will install apache as a dependency.
+	ynh_add_app_dependencies --package="php${phpversion}-fpm"
+	ynh_add_app_dependencies --package="php$phpversion php${phpversion}-common $package"
+
+	# Set php7.0 back as the default version for php-cli.
+	update-alternatives --set php /usr/bin/php7.0
+
+	# Pin this extra repository after packages are installed to prevent sury of doing shit
+	ynh_pin_repo --package="*" --pin="origin \"packages.sury.org\"" --priority=200 --name=extra_php_version
+	ynh_pin_repo --package="php7.0*" --pin="origin \"packages.sury.org\"" --priority=600 --name=extra_php_version --append
+
+	# Advertise service in admin panel
+	yunohost service add php${phpversion}-fpm --log "/var/log/php${phpversion}-fpm.log"
+}
+
+# Remove the specific version of php used by the app.
+#
+# usage: ynh_install_php
+ynh_remove_php () {
+	# Get the version of php used by this app
+	local phpversion=$(ynh_app_setting_get $app phpversion)
+
+	if [ "$phpversion" == "7.0" ] || [ -z "$phpversion" ]
+	then
+		if [ "$phpversion" == "7.0" ]
+		then
+			ynh_print_err "Do not use ynh_remove_php to install php7.0"
+		fi
+		return 0
+	fi
+
+	# Remove the line for this app
+	sed --in-place "/$YNH_APP_INSTANCE_NAME:$phpversion/d" "/etc/php/ynh_app_version"
+
+	# If no other app uses this version of php, remove it.
+	if ! grep --quiet "$phpversion" "/etc/php/ynh_app_version"
+	then
+		# Purge php dependences for this version.
+		ynh_package_autopurge "php$phpversion php${phpversion}-fpm php${phpversion}-common"
+		# Remove the service from the admin panel
+		yunohost service remove php${phpversion}-fpm
+	fi
```

Comment:
You're right.
There was a warning for php7.0, but nothing for php7.3.

In both the install and remove part, we need a condition to exclude the default version.

Reply:
Added to my todo list about those PR.
However, rather prefer to have #880 merged with that PR to fix that problem. Since 880 will add that default php version we need.

Reply:
Should be fixed by https://github.com/YunoHost/yunohost/pull/881/commits/017b0e929c7f6a07f7828013316da7fcc3fe80f5
If the helper is called to remove the default php version, it won't do it.
So, for an app that have installed php7.3 on stretch, after a migration to buster, if the app calls directly ynh_remove_php, the helper will refuse to remove php7.3 as it became the default version for buster.